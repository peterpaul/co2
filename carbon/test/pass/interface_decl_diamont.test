["stdio.h"] int printf(char * format, ...);

interface Foo {
	int baz ();
}

interface Bar {
	int baz ();
}

class FooBar, Foo, Bar {

	int baz (), Foo {
		return 5;
	}

	int barabaz (), Bar.baz {
		return 7;
	}
}

class Baz, Foo, Bar {
	int test (), Foo.baz, Bar.baz {
		return 6;
	}
}

int main () {
	FooBar fooBar = new FooBar ();
	Foo foo = (Foo) fooBar;
	Bar bar = (Bar) fooBar;
	printf ("foo.baz () = %d\n", foo.baz ());
	printf ("bar.baz () = %d\n", bar.baz ());

	Baz fooBaz = (Baz) new Baz ();
	foo = (Foo) fooBaz;
	bar = (Bar) fooBaz;
	printf ("foo.baz () = %d\n", foo.baz ());
	printf ("bar.baz () = %d\n", bar.baz ());

	return 0;
}
