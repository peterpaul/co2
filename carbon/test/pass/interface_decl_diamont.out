foo.baz () = 5
bar.baz () = 7
foo.baz () = 6
bar.baz () = 6
